# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/prosody/no-umask.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Prevent the makefile from running umask as it can ruin your filesystem tree.

  - NoTag <notag@t2sde.org>

--- prosody-0.12.5/certs/GNUmakefile.vanilla	2025-02-28 23:18:44.492516040 +0100
+++ prosody-0.12.5/certs/GNUmakefile	2025-02-28 23:19:01.185695006 +0100
@@ -19,7 +19,7 @@
 		-sha256 -utf8 -config $(firstword $^) -out $@
 
 %.csr: %.cnf
-	umask 0077 && touch $*.key
+	touch $*.key
 	openssl req -new -newkey rsa:$(keysize) -nodes -keyout $*.key \
 		-sha256 -utf8 -config $^ -out $@
 	@chmod 400 $*.key
@@ -28,7 +28,7 @@
 	openssl req -new -key $^ -utf8 -subj /CN=$* -out $@
 
 %.csr:
-	umask 0077 && touch $*.key
+	touch $*.key
 	openssl req -new -newkey rsa:$(keysize) -nodes -keyout $*.key \
 		-utf8 -subj /CN=$* -out $@
 	@chmod 400 $*.key
@@ -39,7 +39,7 @@
 		-config $(firstword $^) -out $@
 
 %.crt: %.cnf
-	umask 0077 && touch $*.key
+	touch $*.key
 	openssl req -new -x509 -newkey rsa:$(keysize) -nodes -keyout $*.key \
 		-days 365 -sha256 -utf8 -config $(firstword $^) -out $@
 	@chmod 400 $*.key
@@ -48,7 +48,7 @@
 	openssl req -new -x509 -key $^ -days 365 -sha256 -utf8 -subj /CN=$* -out $@
 
 %.crt:
-	umask 0077 && touch $*.key
+	touch $*.key
 	openssl req -new -x509 -newkey rsa:$(keysize) -nodes -keyout $*.key \
 		-days 365 -sha256 -out $@ -utf8 -subj /CN=$*
 	@chmod 400 $*.key
@@ -58,7 +58,7 @@
 	sed 's,example\.com,$*,g' openssl.cnf > $@
 
 %.key:
-	umask 0077 && openssl genrsa -out $@ $(keysize)
+	openssl genrsa -out $@ $(keysize)
 	@chmod 400 $@
 
 # Generate Diffie-Hellman parameters
